R/object_usage_linter.R

```diff
@@ -188,31 +188,26 @@ get_assignment_symbols <- function(xml) {
 }
 
 get_function_assignments <- function(xml) {
-  left_assignment_functions <-
-    xml2::xml_find_all(xml, "expr[LEFT_ASSIGN][expr[2][FUNCTION]]/expr[2]")
-  equal_assignment_functions <-
-    xml2::xml_find_all(xml, "equal_assign[expr[2]][expr[FUNCTION]]/expr[2]")
-  assign_assignment_functions <-
-    xml2::xml_find_all(xml, "expr[expr[SYMBOL_FUNCTION_CALL/text()='assign']]/expr[3]")
-  setmethod_assignment_functions <-
-    xml2::xml_find_all(xml, "expr[expr[SYMBOL_FUNCTION_CALL/text()='setMethod']]/expr[3]")
+  direct_assignment_functions <-
+    xml2::xml_find_all(xml, "//*[((self::expr and LEFT_ASSIGN) or (self::expr_or_assign_or_help and EQ_ASSIGN)) and expr[2][FUNCTION]]/expr[2]")
```

Comment:
we could also consider combining the two xpaths here to avoid calling `xml_find_all()` twice, but for now I left it for readability.